docs: adding details on how to declare a UIB app and other improvements

marcobonita opened the pull request:
Following hours of debug, adding these details will make sure customers won't get stuck like I did

marcobonita committed:
Update package-and-deploy-your-application.adoc

details about JSON file and removal of the "enterprise only" note, since it's redundant

marcobonita committed:
Update builder-declare-interface-in-bonita.adoc

adding mandatory details about token's value

marcobonita committed:
Update production-packaging.adoc

marcobonita committed:
Update package-and-deploy-your-application.adoc

marcobonita committed:
Update package-and-deploy-your-application.adoc

marcobonita committed:
Update package-and-deploy-your-application.adoc

formatting

akantcheff commented on modules/applications/pages/builder-declare-interface-in-bonita.adoc:
```diff
@@ -23,6 +23,10 @@ In this example, the _General Overview_ application is declared with the tag:
 
 There are fewer properties to set for an application link than for a legacy application, but the `displayName` is still mandatory.
 
+The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. It is usually identical to your application's name.
```
> It is usually identical to your application's name.

Not really, the name is converted to kebab-case. So in your example "demo_application" was converted to "demo-application". Another example is "Demo Application" will be converted to "demo-application" as well.
To prevent confusion, I would not say it is identical to the application name.

marcobonita replied:
```suggestion
The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. For example, if your application name is _Awesome application_, its `slug` should be _awesome-application_.
```

Looks better?

akantcheff commented on modules/applications/pages/package-and-deploy-your-application.adoc:
```diff
@@ -62,3 +58,16 @@ The application is on error and the Bonita UI Builder is not able to import and
+[.troubleshooting-section]
+--
+[.symptom]
+After building and running the Bonita UI Builder, no application can't be seen or found in production.
```
```suggestion
After building and running the Bonita UI Builder, no application can be seen or found in production.
```

akantcheff commented on modules/applications/pages/package-and-deploy-your-application.adoc:
```diff
@@ -62,3 +58,16 @@ The application is on error and the Bonita UI Builder is not able to import and
+
+[.solution]#Solution#
+Make sure the application designed with Bonita UI Builder is declared using the <advancedLink> tag. Then, save and deploy the Application Descriptor by clicking the `deploy` button in the Studio.
+Also, make sure the `token` value is identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of the application]. If several `slug` keys are found in the JSON export, you should take the value of the first `slug` key. It is usually identical to your application's name.
```
> It is usually identical to your application's name.

Same feedback

akantcheff commented on modules/applications/pages/production-packaging.adoc:
```diff
@@ -63,3 +75,5 @@ To run the production docker image, you can use the following command:
 
 You can override and configure all xref:ui-builder-docker-installation.adoc#environment-variables[environment variables].
+
+NOTE: If you want to test your applications in production, instead of using `docker run`, you can use a `docker-compose.yml` file in the likes of what was done in the xref:download-and-launch.adoc#_create_the_docker_environment[first steps] (i.e. with a `nginx` reverse proxy and the docker image xref:production-packaging.adoc#_build_docker_image[built previously]).
```
> If you want to test your applications in production

"If you want to test your applications **before going into** production" no?

marcobonita replied:
Correct, changes made

marcobonita committed:
Update modules/applications/pages/package-and-deploy-your-application.adoc

Co-authored-by: Adrien Kantcheff <5028967+akantcheff@users.noreply.github.com>

marcobonita committed:
Update production-packaging.adoc

marcobonita committed:
Update package-and-deploy-your-application.adoc

updated slug details

marcobonita committed:
Merge branch '2024.3' into patch-29

marcobonita committed:
Update modules/applications/pages/builder-declare-interface-in-bonita.adoc

marcobonita committed:
Merge branch '2024.3' into patch-29

marcobonita committed:
Update faq.adoc

removing info about templates (since they're disabled) and changing the info about datasources (part of the "lighten the docker image" project)

marcobonita committed:
Merge branch '2024.3' into patch-29

akantcheff commented on modules/applications/pages/builder-declare-interface-in-bonita.adoc:
```diff
@@ -23,6 +23,10 @@ In this example, the _General Overview_ application is declared with the tag:
 
 There are fewer properties to set for an application link than for a legacy application, but the `displayName` is still mandatory.
 
+The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. For example, if your application name is _Awesome application_, its `slug` should be _awesome-application_.
```
Italic syntax is not correctly formatted
```suggestion
The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. For example, if your application name is __Awesome application__, its `slug` should be __awesome-application__.
```

akantcheff commented on modules/applications/pages/package-and-deploy-your-application.adoc:
```diff
@@ -62,3 +58,16 @@ The application is on error and the Bonita UI Builder is not able to import and
+This is probably due to an error when the application was xref:builder-declare-interface-in-bonita.adoc[declared] in the Application Descriptor. Bonita UI Builder is therefore unable to find the application's location.
+
+[.solution]#Solution#
+Make sure the application designed with Bonita UI Builder is declared using the <advancedLink> tag. Then, save and deploy the Application Descriptor by clicking the `deploy` button in the Studio.
```
```suggestion
Ensure the application designed with Bonita UI Builder is declared using the `<advancedLink>` tag. Then, save and deploy the Application Descriptor by clicking the `deploy` button in the Studio.
```

marcobonita committed:
Update modules/applications/pages/builder-declare-interface-in-bonita.adoc

Co-authored-by: Adrien Kantcheff <5028967+akantcheff@users.noreply.github.com>

marcobonita committed:
Update modules/applications/pages/package-and-deploy-your-application.adoc

Co-authored-by: Adrien Kantcheff <5028967+akantcheff@users.noreply.github.com>
